ubuntu_image/parser.py

```diff
@@ -283,9 +283,11 @@ def parse(stream_or_string):
             # Extract the rest of the structure data.
             structure_id = structure.get('id')
             filesystem = structure['filesystem']
-            if (structure_type == 'mbr' and
-                    filesystem is not FileSystemType.none):
-                raise ValueError('mbr type must not specify a file system')
+            if structure_type == 'mbr':
+                if structure_id:
```

Comment:
Generally, I prefer to have narrower conditionals, so in this case if the structure id is given it will be `not None`.  I'll make this change through the GH u/i.  Other than that it looks great!  Thanks.